src/main/java/com/asem/regestration/model/dao/impl/CourseDAOImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.asem.regestration.model.dao.impl;
+
+import com.asem.regestration.model.dao.CourseDAO;
+import com.asem.regestration.model.dao.utils.DataSourcePool;
+import com.asem.regestration.model.entity.Courses;
+import com.asem.regestration.model.entity.Student;
+
+import java.sql.*;
+
+public class CourseDAOImpl implements CourseDAO {
+    private String userName;
+    private String passwored;
+    private String host;
+    private String databaseName;
+    private DataSourcePool dataSourcePool;
+
+
+    public CourseDAOImpl(String userName, String passwored, String host, String databaseName) {
+        this.userName=userName;
+        this.passwored=passwored;
+        this.host=host;
+        this.databaseName=databaseName;
+        this.dataSourcePool=new DataSourcePool(userName,passwored,host,databaseName);
+
+    }
+    public Connection getConnection() throws SQLException {
```

Comment:
Instead of returning connection and handle it manually in each method you can return the prepareStatement itself